emr-user-role-mapper-application/src/test/com/amazon/aws/emr/integration/utils/IAMUtils.java

```diff
@@ -0,0 +1,87 @@
+package com.amazon.aws.emr.integration.utils;
+
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagement;
+import com.amazonaws.services.identitymanagement.AmazonIdentityManagementClientBuilder;
+import com.amazonaws.services.identitymanagement.model.AttachRolePolicyRequest;
+import com.amazonaws.services.identitymanagement.model.CreatePolicyRequest;
+import com.amazonaws.services.identitymanagement.model.CreatePolicyResult;
+import com.amazonaws.services.identitymanagement.model.CreateRoleRequest;
+import com.amazonaws.services.identitymanagement.model.CreateRoleResult;
+import com.amazonaws.services.identitymanagement.model.DeletePolicyRequest;
+import com.amazonaws.services.identitymanagement.model.DeleteRoleRequest;
+import com.amazonaws.services.identitymanagement.model.DetachRolePolicyRequest;
+import com.amazonaws.services.identitymanagement.model.EntityAlreadyExistsException;
+import com.amazonaws.services.identitymanagement.model.GetPolicyRequest;
+import com.amazonaws.services.identitymanagement.model.GetRoleRequest;
+import com.amazonaws.services.identitymanagement.model.Policy;
+import com.amazonaws.services.identitymanagement.model.Role;
+
+/**
+ * Utility methods for AWS IAM.
+ */
+public class IAMUtils {
+
+  final static AmazonIdentityManagement iam =
+      AmazonIdentityManagementClientBuilder.defaultClient();
+
+
+  public static Policy createPolicy(String account, String policy) {
+    String policyName = "urm-policy" + policy.hashCode();
+    CreatePolicyResult response;
+    try {
+      CreatePolicyRequest request = new CreatePolicyRequest()
+          .withPolicyName(policyName)
+          .withPolicyDocument(policy);
+      response = iam.createPolicy(request);
+    } catch (EntityAlreadyExistsException ex) {
+      String policyArn = getIamPolicyArn(account, policyName);
+      return iam.getPolicy(new GetPolicyRequest().withPolicyArn(policyArn)).getPolicy();
+    }
+    return response.getPolicy();
+  }
+
+  private static String getIamPolicyArn(String account, String policyName) {
+    return String.format("arn:aws:iam::%s:policy/%s", account, policyName);
+  }
+
+  public static void attachPolicyToRole(String roleArn, String PolicyArn) {
+    AttachRolePolicyRequest attach_request =
+        new AttachRolePolicyRequest()
+            .withRoleName(roleArn)
+            .withPolicyArn(PolicyArn);
+
+    iam.attachRolePolicy(attach_request);
```

Comment:
What happens if the policy is already attached?

Reply:
We create a new policy on each test (new ARN). So this case should not arise I think.

Reply:
It's idempotent.

```
aws iam attach-role-policy --policy-arn arn:aws:iam::176430881729:policy/urm-policy-107149120 --role-name test-integ-urm-role---871835695--463378646
a483e77e76c2:amazon-emr-user-role-mapper shkala$ aws iam attach-role-policy --policy-arn arn:aws:iam::176430881729:policy/urm-policy-107149120 --role-name test-integ-urm-role---871835695--463378646
```